[package] opencolorio/2.1.0: missing strlen when building package

<!-- 
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.
-->

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **opencolorio/2.1.0**
  * Operating System+version: **Linux Fedora 36**
  * Compiler+version: **GCC 12.1.1**
  * Conan version: **conan 1.48.1**
  * Python version: **Python 3.10.4**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=12
compiler.libcxx=libstdc++11
build_type=Release
[options]
[conf]
[build_requires]
[env]

```


### Steps to reproduce (Include if Applicable)

``conan create . 2.1.0@`` in package folder

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Exporting package recipe
opencolorio/2.1.0 exports: File 'conandata.yml' found. Exporting it...
opencolorio/2.1.0 exports: Copied 1 '.yml' file: conandata.yml
opencolorio/2.1.0: Calling export_sources()
opencolorio/2.1.0 export_sources() method: Copied 1 '.txt' file: CMakeLists.txt
opencolorio/2.1.0 export_sources() method: Copied 2 '.patch' files: fix-cmake-source-dir-and-targets.patch, pstring.patch
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'2.1.0': {'url': 'https://github.com/AcademySoftwareFoundation/OpenColorIO/archive/v2.1.0.tar.gz', 'sha256': '81fc7853a490031632a69c73716bc6ac271b395e2ba0e2587af9995c2b0efb5f'}}, 'patches': {'2.1.0': [{'patch_file': 'patches/fix-cmake-source-dir-and-targets.patch', 'base_path': 'source_subfolder'}, {'patch_file': 'patches/pstring.patch', 'base_path': 'source_subfolder'}]}}
[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] New conandata.yml contents: patches:
  2.1.0:
  - base_path: source_subfolder
    patch_file: patches/fix-cmake-source-dir-and-targets.patch
  - base_path: source_subfolder
    patch_file: patches/pstring.patch
sources:
  2.1.0:
    sha256: 81fc7853a490031632a69c73716bc6ac271b395e2ba0e2587af9995c2b0efb5f
    url: https://github.com/AcademySoftwareFoundation/OpenColorIO/archive/v2.1.0.tar.gz

[HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] OK
[HOOK - conan-center.py] post_export(): [DEFAULT SHARED OPTION VALUE (KB-H050)] OK
[HOOK - conan-center.py] post_export(): [LICENSE PUBLIC DOMAIN (KB-H056)] OK
opencolorio/2.1.0: A new conanfile.py version was exported
opencolorio/2.1.0: Folder: /home/anton/.conan/data/opencolorio/2.1.0/_/_/export
opencolorio/2.1.0: Exported revision: dd7155a6c074c79e148dc6d77922b730
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

opencolorio/2.1.0 (test package): Installing package
Requirements
    expat/2.4.8 from 'conancenter' - Cache
    lcms/2.13.1 from 'conancenter' - Cache
    opencolorio/2.1.0 from local cache - Cache
    openexr/2.5.7 from 'conancenter' - Cache
    pystring/1.1.3 from 'conancenter' - Cache
    yaml-cpp/0.7.0 from 'conancenter' - Cache
    zlib/1.2.12 from 'conancenter' - Cache
Packages
    expat/2.4.8:a1ba772dbe13186199f7ecf8dde362ba9613ecf7 - Cache
    lcms/2.13.1:2a19826344ff00be1c04403f2f8e7008ed3a7cc6 - Cache
    opencolorio/2.1.0:53a2a11b1af28774b114d4f0b11e1127e3bd7ec6 - Build
    openexr/2.5.7:83df7a6c3cca507b416be2520dcbf0d034646b90 - Cache
    pystring/1.1.3:174df609440df17baf334e1f5953cca909f2ae18 - Cache
    yaml-cpp/0.7.0:174df609440df17baf334e1f5953cca909f2ae18 - Cache
    zlib/1.2.12:2a19826344ff00be1c04403f2f8e7008ed3a7cc6 - Cache

Installing (downloading, building) binaries...
expat/2.4.8: Already installed!
lcms/2.13.1: Already installed!
pystring/1.1.3: Already installed!
yaml-cpp/0.7.0: Already installed!
zlib/1.2.12: Already installed!
openexr/2.5.7: Already installed!
[HOOK - conan-center.py] pre_source(): [IMMUTABLE SOURCES (KB-H010)] OK
opencolorio/2.1.0: Configuring sources in /home/anton/.conan/data/opencolorio/2.1.0/_/_/source
Downloading v2.1.0.tar.gz completed [10713.72k]                                          

[HOOK - conan-center.py] post_source(): [LIBCXX MANAGEMENT (KB-H011)] OK
[HOOK - conan-center.py] post_source(): [CPPSTD MANAGEMENT (KB-H022)] OK
[HOOK - conan-center.py] post_source(): [SHORT_PATHS USAGE (KB-H066)] OK
opencolorio/2.1.0: Copying sources to build folder
opencolorio/2.1.0: Building your package in /home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6
opencolorio/2.1.0: Generator cmake created conanbuildinfo.cmake
opencolorio/2.1.0: Generator cmake_find_package created FindEXPAT.cmake
opencolorio/2.1.0: Generator cmake_find_package created FindOpenEXR.cmake
opencolorio/2.1.0: Generator cmake_find_package created Findyaml-cpp.cmake
opencolorio/2.1.0: Generator cmake_find_package created Findpystring.cmake
opencolorio/2.1.0: Generator cmake_find_package created Findlcms.cmake
opencolorio/2.1.0: Generator cmake_find_package created FindZLIB.cmake
opencolorio/2.1.0: Aggregating env generators
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK. 'fPIC' option found and apparently well managed
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
opencolorio/2.1.0: Calling build()
-- The C compiler identification is GNU 12.1.1
-- The CXX compiler identification is GNU 12.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- Conan: Compiler GCC>=5, checking major version 12
-- Conan: Checking correct version: 12
-- Conan: C++ stdlib: libstdc++11
-- Setting C++ version to '11' as none was specified.
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Found OpenGL: /usr/lib64/libOpenGL.so  found components: OpenGL 
-- Could NOT find GLEW (missing: GLEW_INCLUDE_DIRS GLEW_LIBRARIES) 
-- Use "GLEW_ROOT" to specify an install location
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Checking for module 'glut'
--   Package 'glut', required by 'virtual:world', not found
-- Could NOT find GLUT (missing: GLUT_glut_LIBRARY GLUT_INCLUDE_DIR) 
-- Use "GLUT_ROOT" to specify an install location
CMake Warning at source_subfolder/share/cmake/utils/CheckSupportGL.cmake:47 (message):
  GPU rendering disabled
Call Stack (most recent call first):
  source_subfolder/CMakeLists.txt:140 (include)


-- Performing Test HAVE_SSE2
-- Performing Test HAVE_SSE2 - Success
-- Setting SOVERSION to '2.1' as none was specified.
-- Conan: Using autogenerated FindEXPAT.cmake
-- Found EXPAT: 2.4.8 (found suitable version "2.4.8", minimum required is "2.2.8") 
-- Library expat found /home/anton/.conan/data/expat/2.4.8/_/_/package/a1ba772dbe13186199f7ecf8dde362ba9613ecf7/lib/libexpat.a
-- Found: /home/anton/.conan/data/expat/2.4.8/_/_/package/a1ba772dbe13186199f7ecf8dde362ba9613ecf7/lib/libexpat.a
-- Conan: Using autogenerated Findyaml-cpp.cmake
-- Found yaml-cpp: 0.7.0 (found suitable version "0.7.0", minimum required is "0.6.3") 
-- Library yaml-cpp found /home/anton/.conan/data/yaml-cpp/0.7.0/_/_/package/174df609440df17baf334e1f5953cca909f2ae18/lib/libyaml-cpp.a
-- Found: /home/anton/.conan/data/yaml-cpp/0.7.0/_/_/package/174df609440df17baf334e1f5953cca909f2ae18/lib/libyaml-cpp.a
-- Conan: Using autogenerated Findpystring.cmake
-- Found pystring: 1.1.3 (found suitable version "1.1.3", minimum required is "1.1.3") 
-- Library pystring found /home/anton/.conan/data/pystring/1.1.3/_/_/package/174df609440df17baf334e1f5953cca909f2ae18/lib/libpystring.a
-- Found: /home/anton/.conan/data/pystring/1.1.3/_/_/package/174df609440df17baf334e1f5953cca909f2ae18/lib/libpystring.a
-- Conan: Using autogenerated FindOpenEXR.cmake
-- Found OpenEXR: 2.5.7 (found suitable version "2.5.7", minimum required is "2.4.0") 
-- Library IlmImfUtil-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImfUtil-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImfUtil-2_5.a
-- Library IlmImf-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImf-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImf-2_5.a
-- Library Imath-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libImath-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libImath-2_5.a
-- Library Half-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libHalf-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libHalf-2_5.a
-- Library IexMath-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIexMath-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIexMath-2_5.a
-- Library IlmThread-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmThread-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmThread-2_5.a
-- Library Iex-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIex-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIex-2_5.a
-- Conan: Using autogenerated FindZLIB.cmake
-- Found ZLIB: 1.2.12 (found version "1.2.12") 
-- Library z found /home/anton/.conan/data/zlib/1.2.12/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/lib/libz.a
-- Found: /home/anton/.conan/data/zlib/1.2.12/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/lib/libz.a
-- Library Iex-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIex-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIex-2_5.a
-- Library IlmThread-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmThread-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmThread-2_5.a
-- Library IexMath-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIexMath-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIexMath-2_5.a
-- Library Half-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libHalf-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libHalf-2_5.a
-- Library Imath-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libImath-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libImath-2_5.a
-- Library IlmImf-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImf-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImf-2_5.a
-- Library IlmImfUtil-2_5 found /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImfUtil-2_5.a
-- Found: /home/anton/.conan/data/openexr/2.5.7/_/_/package/83df7a6c3cca507b416be2520dcbf0d034646b90/lib/libIlmImfUtil-2_5.a
-- Conan: Using autogenerated Findlcms.cmake
-- Found lcms: 2.13.1 (found suitable version "2.13.1", minimum required is "2.2") 
-- Library lcms2 found /home/anton/.conan/data/lcms/2.13.1/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/lib/liblcms2.a
-- Found: /home/anton/.conan/data/lcms/2.13.1/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/lib/liblcms2.a
-- Could NOT find OpenImageIO (missing: OPENIMAGEIO_INCLUDE_DIR OPENIMAGEIO_LIBRARY OPENIMAGEIO_VERSION) (Required is at least version "2.1.9")
CMake Warning at source_subfolder/tests/CMakeLists.txt:46 (message):
  Could NOT find OpenImageIO.  Skipping build the OSL unit tests


CMake Warning at source_subfolder/src/apps/ociochecklut/CMakeLists.txt:5 (message):
  GL component missing.  GPU disabled for ociochecklut.


CMake Warning at source_subfolder/src/apps/CMakeLists.txt:17 (message):
  Could NOT find OpenImageIO.  Skipping build for ociolutimage, ocioconvert,
  ociodisplay and ocioperf


CMake Warning at source_subfolder/src/libutils/oglapphelpers/CMakeLists.txt:5 (message):
  GL component missing.  Skipping oglapphelpers.


-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    OCIO_INSTALL_EXT_PACKAGE
    OCIO_USE_BOOST_PTR


-- Build files have been written to: /home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6/build_subfolder
[  1%] Building CXX object source_subfolder/src/apputils/CMakeFiles/apputils.dir/strutil.cpp.o
[  1%] Building CXX object source_subfolder/src/apputils/CMakeFiles/apputils.dir/argparse.cpp.o
[  1%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/apphelpers/CategoryHelpers.cpp.o
[  2%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/apphelpers/ColorSpaceHelpers.cpp.o
[  2%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/apphelpers/DisplayViewHelpers.cpp.o
[  2%] Linking CXX static library ../../../lib/libapputils.a
[  2%] Built target apputils
[  3%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/apphelpers/LegacyViewingPipeline.cpp.o
[  3%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/apphelpers/MixingHelpers.cpp.o
[  4%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Baker.cpp.o
[  4%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/BitDepthUtils.cpp.o
[  5%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Caching.cpp.o
[  6%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/ColorSpace.cpp.o
[  6%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/ColorSpaceSet.cpp.o
[  7%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Config.cpp.o
[  7%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Context.cpp.o
[  8%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/ContextVariableUtils.cpp.o
[  8%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/CPUProcessor.cpp.o
[  9%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Display.cpp.o
[  9%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/DynamicProperty.cpp.o
[ 10%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Exception.cpp.o
[ 11%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/cdl/CDLParser.cpp.o
[ 11%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/cdl/CDLReaderHelper.cpp.o
[ 12%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/cdl/CDLWriter.cpp.o
[ 12%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/ctf/CTFReaderHelper.cpp.o
[ 13%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/ctf/CTFReaderUtils.cpp.o
[ 13%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/ctf/CTFTransform.cpp.o
[ 14%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/ctf/IndexMapping.cpp.o
[ 14%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormat3DL.cpp.o
[ 15%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatCCC.cpp.o
[ 16%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatCC.cpp.o
[ 16%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatCDL.cpp.o
[ 17%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatCSP.cpp.o
[ 17%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatCTF.cpp.o
[ 18%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatDiscreet1DL.cpp.o
[ 18%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatHDL.cpp.o
[ 19%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatICC.cpp.o
[ 19%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatIridasCube.cpp.o
[ 20%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatIridasItx.cpp.o
[ 21%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatIridasLook.cpp.o
[ 21%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatPandora.cpp.o
[ 22%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatResolveCube.cpp.o
[ 22%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatSpi1D.cpp.o
[ 23%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatSpi3D.cpp.o
[ 23%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatSpiMtx.cpp.o
[ 24%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatTruelight.cpp.o
[ 24%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatUtils.cpp.o
[ 25%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FileFormatVF.cpp.o
[ 26%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/FormatMetadata.cpp.o
[ 26%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/xmlutils/XMLReaderHelper.cpp.o
[ 27%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/xmlutils/XMLReaderUtils.cpp.o
[ 27%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/fileformats/xmlutils/XMLWriterUtils.cpp.o
[ 28%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/FileRules.cpp.o
[ 28%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/GPUProcessor.cpp.o
/home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6/source_subfolder/src/OpenColorIO/FileRules.cpp: В функции «std::string OpenColorIO_v2_1::{anonymous}::ConvertToRegularExpression(const char*, bool)»:
/home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6/source_subfolder/src/OpenColorIO/FileRules.cpp:64:31: ошибка: нет декларации «strlen» в этой области видимости
   64 |         const size_t length = strlen(globPattern);
      |                               ^~~~~~
/home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6/source_subfolder/src/OpenColorIO/FileRules.cpp:18:1: замечание: «strlen» is defined in header «<cstring>»; did you forget to «#include <cstring>»?
   17 | #include "utils/StringUtils.h"
  +++ |+#include <cstring>
   18 | 
[ 29%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/GpuShader.cpp.o
[ 29%] Building CXX object source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/GpuShaderDesc.cpp.o
gmake[2]: *** [source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/build.make:748: source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/FileRules.cpp.o] Ошибка 1
gmake[2]: *** Ожидание завершения заданий…
gmake[1]: *** [CMakeFiles/Makefile2:583: source_subfolder/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/all] Ошибка 2
gmake: *** [Makefile:136: all] Ошибка 2
opencolorio/2.1.0: 
opencolorio/2.1.0: ERROR: Package '53a2a11b1af28774b114d4f0b11e1127e3bd7ec6' build failed
opencolorio/2.1.0: WARN: Build folder /home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6
ERROR: opencolorio/2.1.0: Error in build() method, line 121
        cm.build()
        ConanException: Error 2 while executing cmake --build '/home/anton/.conan/data/opencolorio/2.1.0/_/_/build/53a2a11b1af28774b114d4f0b11e1127e3bd7ec6/build_subfolder' '--' '-j4'

```

</details>

--- a/recipes/opencolorio/all/conandata.yml
+++ b/recipes/opencolorio/all/conandata.yml
@@ -11,6 +11,8 @@ patches:
       base_path: "source_subfolder"
     - patch_file: "patches/pstring.patch"
       base_path: "source_subfolder"
+    - patch_file: "patches/strlen.patch"
+      base_path: "source_subfolder"
   "1.1.1":
     - patch_file: "patches/1.1.1.patch"
       base_path: "source_subfolder"

--- a/recipes/opencolorio/all/patches/strlen.patch
+++ b/recipes/opencolorio/all/patches/strlen.patch
@@ -6,7 +6,7 @@ index 794dfdbe..94729459 100644
  #include <map>
  #include <regex>
  #include <sstream>
-+#include <cstring>
++#include <string.h>
  
  #include <OpenColorIO/OpenColorIO.h>
  

--- a/recipes/opencolorio/all/patches/strlen.patch
+++ b/recipes/opencolorio/all/patches/strlen.patch
@@ -1,0 +1,12 @@
+diff --git a/src/OpenColorIO/FileRules.cpp b/src/OpenColorIO/FileRules.cpp
+index 794dfdbe..94729459 100644
+--- a/src/OpenColorIO/FileRules.cpp
++++ b/src/OpenColorIO/FileRules.cpp
+@@ -6,6 +6,7 @@
+ #include <map>
+ #include <regex>
+ #include <sstream>
++#include <cstring>
+ 
+ #include <OpenColorIO/OpenColorIO.h>
+ 

